tests/unit/acquisition/function/test_active_learning.py

```diff
@@ -393,6 +398,41 @@ def test_integrated_variance_reduction_builder_updates_without_retracing() -> No
     assert acq_fn.__call__._get_tracing_count() == 1  # type: ignore
 
 
+@pytest.mark.parametrize(
+    "at",
+    [
+        (tf.constant([[[-1.0]]])),
+        (tf.constant([[-0.5]])),
+        (tf.constant([[0.0]])),
+        (tf.constant([[0.5]])),
+        (tf.constant([[1.0]])),
+    ],
+)
+def test_bayesian_active_learning_by_disagreement(at: tf.Tensor) -> None:
```

Comment:
Not sure if this is the right way to test this, or there are better strategies ? @hstojic

Reply:
reimplementing equations again is not a way to go, test should try to verify them independently - partly if not fully... if its not really feasible then we rely on integration tests

here I think we can test it independently actually:
1. partial test, for example, it seems like 2nd term can be checked in a different way, see on page 5 this paragraph "Fig. 1 depicts the striking accuracy ..." 
2. partial test again, for the second term only, but probably better MC based test, see "Quantifying Approximation Losses" in section 5

Reply:
Just replace it with MC based test for both term1 and term2, could you please verify if the code for term2 is indeed right? 
thanks for your patience!